core/backend/src/test/ElementDrivesElement.test.ts

```diff
@@ -0,0 +1,947 @@
+    }
+    HubMock.shutdown();
+  });
+  it("local: topological sort", async () => {
+    const graph = new Graph<string>();
+    // Graph structure:
+    //   1
+    //  / \
+    // 2   3
+    // |\   \
+    // | \   \
+    // 4  5   4
+    //  \     /
+    //   \   /
+    //     5
+
+    graph.addEdge("1", ["2", "3"]);
+    graph.addEdge("2", ["5", "4"]);
+    graph.addEdge("3", ["4"]);
+    graph.addEdge("4", ["5"]);
+    const df = TopologicalSorter.sortDepthFirst(graph);
+    chai.expect(TopologicalSorter.validate(graph, df)).to.be.true;
+    chai.expect(df).to.deep.equal(["1", "3", "2", "4", "5"]);
+
+    const bf = TopologicalSorter.sortBreadthFirst(graph);
+    chai.expect(TopologicalSorter.validate(graph, bf)).to.be.true;
+    chai.expect(bf).to.deep.equal(["1", "2", "3", "4", "5"]);
+  });
+
+  it("local: cycle detection (suppress cycles)", async () => {
+    const graph = new Graph<string>();
+    // Graph structure:
+    // 1 --> 2 <-- 3
+    // ^           |
+    // |-----------|
+    graph.addEdge("1", ["2"]);
+    graph.addEdge("2", ["3"]);
+    graph.addEdge("3", ["1"]);
+
+    const df = TopologicalSorter.sortDepthFirst(graph, [], false);
+    chai.expect(TopologicalSorter.validate(graph, df)).to.be.false;
+    chai.expect(df).to.deep.equal(["1", "2", "3"]);
+
+    // const bf = TopologicalSorter.sortBreadthFirst(graph, [], false);
+    // chai.expect(TopologicalSorter.validate(graph, bf)).to.be.false;
+    // chai.expect(bf).to.deep.equal(["1", "2", "3"]);
+  });
+  it("local: cycle detection (throw)", async () => {
+    const graph = new Graph<string>();
+    // Graph structure:
+    // A---B---C
+    //.  \ D /
+    // 1 --> 2 --> 3
+    // ^           |
+    // |-----------|
+    graph.addEdge("1", ["2"]);
+    graph.addEdge("2", ["3"]);
+    graph.addEdge("3", ["1"]);
+
+    chai.expect(() => TopologicalSorter.sortDepthFirst(graph)).to.throw(
+      "Graph has a cycle"
+    );
+  });
+  it("EDE/local: build system dependencies", async () => {
+    const graph = new Graph<string>();
+    /*
+      Example: Build system dependencies
+
+      - Compile main.c and util.c to main.o and util.o
+      - Link main.o and util.o to produce app.exe
+      - app.exe depends on config.json
+      - test.exe depends on main.o, util.o, and test.c
+
+      Graph:
+        main.c   util.c   test.c   config.json
+          |        |        |           |
+          v        v        |           |
+        main.o   util.o     |           |
+          \      /          |           |
+            \  /            |           |
+           app.exe        test.exe      |
+             |                |         |
+             +----------------+---------+
+    */
+
+    graph.addEdge("main.c", ["main.o"]);
+    graph.addEdge("util.c", ["util.o"]);
+    graph.addEdge("test.c", ["test.exe"]);
+    graph.addEdge("main.o", ["app.exe", "test.exe"]);
+    graph.addEdge("util.o", ["app.exe", "test.exe"]);
+    graph.addEdge("config.json", ["app.exe"]);
```

Comment:
the digram shows config.json connected to both app.exe and test.exe but the doc says app.exe depends on it.  graph matches the docs not the graph.